### PR TITLE
(CDAP-13234) Added a common base class for consuming messages from TMS

### PR DESCRIPTION
**Pull request by chtyim**
- It handle failures and keep tracks of states

### CODE REVIEW

File: cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/package-info.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * This package contains classes to help writing high level consumers that need to deal with failure
+ * and persisting of message id.
+ */
+package co.cask.cdap.messaging.subscriber;
```
Comment (albertshau) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/package-info.java]: does this file need a newline?

**Review by albertshau — APPROVED** [review]
one small question, otherwise lgtm

File: cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/package-info.java (hunk 1 above)
Comment (chtyim) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/package-info.java]: Updated

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT6370

**Pull request merged**